src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/DynamicVariantExtensions.cs

```diff
@@ -66,7 +66,7 @@ public static void SetAsIConvertible(this ref ComVariant variant, IConvertible v
                 case TypeCode.Int32: variant = ComVariant.Create(value.ToInt32(ci)); break;
                 case TypeCode.UInt32: variant = ComVariant.Create(value.ToUInt32(ci)); break;
                 case TypeCode.Int64: variant = ComVariant.Create(value.ToInt64(ci)); break;
-                case TypeCode.UInt64: variant = ComVariant.Create(value.ToInt64(ci)); break;
```

Comment:
This may be a real bug

Reply:
This is a long-standing bug from at least .NET 6, likely .NET Framework (as it's in the DLR version this code was initially ported from as well): https://github.com/dotnet/runtime/blob/f1dd57165bfd91875761329ac3a8b17f6606ad18/src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/Variant.Extended.cs#L307

I think we can fix this going forward, but as this does change the variant type I'd be concerned of breaking consumers if we were to backport this fix.

Reply:
Right, I do not think this is something that we want to backport.